jdaviz/configs/default/plugins/viewers.py

```diff
@@ -174,6 +174,11 @@ def _on_layers_update(self, layers=None):
                     self._expected_subset_layers.remove(layer.layer.label)
                 self._expected_subset_layer_default(layer)
 
+        # We need this so that ShadowSpatialSpectral marks are removed
+        # when data is removed
+        if self.__class__.__name__ == 'CubevizProfileView':
+            self._check_if_data_removed_for_spatial_spectral()
```

Comment:
can we do this by subscribing to a message instead and possibly merge with `_on_subset_delete`?